src/brucke_health_check.erl

```diff
@@ -0,0 +1,131 @@
+%%%
+%%%   Copyright (c) 2017 Klarna AB
+%%%
+%%%   Licensed under the Apache License, Version 2.0 (the "License");
+%%%   you may not use this file except in compliance with the License.
+%%%   You may obtain a copy of the License at
+%%%
+%%%       http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%%   Unless required by applicable law or agreed to in writing, software
+%%%   distributed under the License is distributed on an "AS IS" BASIS,
+%%%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%%   See the License for the specific language governing permissions and
+%%%   limitations under the License.
+%%%
+-module(brucke_health_check).
+
+-define(UPSTREAM, <<"upstream">>).
+-define(DOWNSTREAM, <<"downstream">>).
+-define(PRODUCERS, <<"producers">>).
+-define(CONSUMERS, <<"consumers">>).
+-define(STATUS, <<"status">>).
+-define(MEMBERS, <<"members">>).
+-define(TOPIC, <<"topic">>).
+-define(ID, <<"id">>).
+-define(PARTITION, <<"partition">>).
+
+-include("brucke_int.hrl").
+
+%% API
+-export([init/0, get_report_data/1]).
+
+init() ->
+  case application:get_env(?APPLICATION, healthcheck) of
+    {ok, true} ->
+      Port = application:get_env(?APPLICATION, healthcheck_port, 8080),
+      Dispatch = cowboy_router:compile([{'_', [{'_', brucke_http_handler, []}]}]),
+      {ok, _} = cowboy:start_clear(http, 1, [{port, Port}], #{env => #{dispatch => Dispatch}}),
+      ok;
+    _ -> ok
+  end.
+
+-spec get_report_data(proplists:proplist()) -> map().
+get_report_data(Params) ->
+  AskClients = proplists:get_value(<<"clients">>, Params, true),
+  AllCliends = get_all_clients(AskClients),
+  AskRoutes = proplists:get_value(<<"routes">>, Params, true),
+  AllRoutes = get_all_routes(AskRoutes),
+  #{<<"clients">> => AllCliends, <<"routes">> => AllRoutes}.
+
+
+%% @private
+-spec get_all_clients(boolean()) -> list(map()).
+get_all_clients(true) ->
+  AllClients = brucke_config:all_clients(),
+  lists:map(
+    fun({ClientId, _, _}) ->
+      case whereis(ClientId) of
+        undefined -> #{ClientId => undefined};
+        _ -> #{ClientId => ok}
+      end
+    end, AllClients);
+get_all_clients(_) -> [].
+
+%% @private
+-spec get_all_routes(binary()) -> list(map()).
+get_all_routes(true) ->
+  AllRoutes = brucke_config:all_routes(),
```

Comment:
this is the result taken from ets table.
if a route is conigured wrong in yml, bruke will start, but with the bad route discarded.
(an 'ALERT' level log is emitted at start)

We should perhaps also keep a record of the discarded routes (for example in another ets)